element_volume/volume.py

```diff
@@ -0,0 +1,261 @@
+
+        self.insert1(
+            dict(
+                **key,
+                volume=volume_data,
+                px_width=volume_data.shape[2],
+                px_height=volume_data.shape[1],
+                px_depth=volume_data.shape[0]
+            )
+        )
+
+
+@schema
+class SegmentationParamset(dj.Lookup):
+    definition = """
+    paramset_idx: int
+    ---
+    segmentation_method: varchar(32)
+    paramset_desc="": varchar(256)
+    params: longblob
+    paramset_hash: uuid
+    unique index (paramset_hash)
+    """
+
+    @classmethod
+    def insert_new_params(
+        cls,
+        segmentation_method: str,
+        paramset_desc: str = "",
+        params: dict = {},
+        paramset_idx: int = None,
+    ):
+        """Inserts new parameters into the table.
+
+        Args:
+            segmentation_method (str): name of the clustering method.
+            paramset_desc (str): description of the parameter set
+            params (dict): clustering parameters
+            paramset_idx (int, optional): Unique parameter set ID. Defaults to None.
+        """
+        if paramset_idx is None:
+            paramset_idx = (
+                dj.U().aggr(cls, n="max(paramset_idx)").fetch1("n") or 0
+            ) + 1
+
+        param_dict = {
+            "segmentation_method": segmentation_method,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "paramset_idx": paramset_idx,
+            "paramset_hash": dict_to_uuid(
+                {**params, "segmentation_method": segmentation_method}
+            ),
+        }
+        param_query = cls & {"paramset_hash": param_dict["paramset_hash"]}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1("paramset_idx")
+            if (
+                existing_paramset_idx == paramset_idx
+            ):  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    f"The specified param-set already exists"
+                    f" - with paramset_idx: {existing_paramset_idx}"
+                )
+        else:
+            if {"paramset_idx": paramset_idx} in cls.proj():
+                raise dj.DataJointError(
+                    f"The specified paramset_idx {paramset_idx} already exists,"
+                    f" please pick a different one."
+                )
+            cls.insert1(param_dict)
+
+
+@schema
+class SegmentationTask(dj.Manual):
+    definition = """
+    -> Volume
+    -> SegmentationParamset
+    ---
+    segmentation_output_dir='': varchar(255)  #  Output directory of the segmented results relative to root data directory
+    task_mode='load': enum('load', 'trigger')
+    """
+
+
+@schema
+class Segmentation(dj.Computed):
```

Comment:
Cellpose is actually a retrainable deep learning model. This design allows only model evaluation. I'm not suggesting any change at this point, just making a note for future.